File: app/src/main/java/com/brainwellnessspa/invoicemodule/models/InvoiceListModel.java
package com.brainwellnessspa.invoicemodule.models;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

public class InvoiceListModel {
    @SerializedName("ResponseData")
    @Expose
    private ResponseData responseData;
    @SerializedName("ResponseCode")
    @Expose
    private String responseCode;
    @SerializedName("ResponseMessage")
    @Expose
    private String responseMessage;
    @SerializedName("ResponseStatus")
    @Expose
    private String responseStatus;

    public ResponseData getResponseData() {
        return responseData;
    }

    public void setResponseData(ResponseData responseData) {
        this.responseData = responseData;
    }

    public String getResponseCode() {
        return responseCode;
    }

    public void setResponseCode(String responseCode) {
        this.responseCode = responseCode;
    }

    public String getResponseMessage() {
        return responseMessage;
    }

    public void setResponseMessage(String responseMessage) {
        this.responseMessage = responseMessage;
    }

    public String getResponseStatus() {
        return responseStatus;
    }

    public void setResponseStatus(String responseStatus) {
        this.responseStatus = responseStatus;
    }

    public static class ResponseData implements Parcelable {
        @SerializedName("Appointment")
        @Expose
        private ArrayList<Appointment> appointment = null;
        @SerializedName("MemberShip")
        @Expose
        private ArrayList<MemberShip> memberShip = null;

        public static final Creator<ResponseData> CREATOR = new Creator<ResponseData>() {
            @Override
            public ResponseData createFromParcel(Parcel in) {
                return new ResponseData(in);
            }

            @Override
            public ResponseData[] newArray(int size) {
                return new ResponseData[size];
            }
        };

        public ArrayList<Appointment> getAppointment() {
            return appointment;
        }

        public void setAppointment(ArrayList<Appointment> appointment) {
            this.appointment = appointment;
        }

        public ArrayList<MemberShip> getMemberShip() {
            return memberShip;
        }

        public void setMemberShip(ArrayList<MemberShip> memberShip) {
            this.memberShip = memberShip;
        }

        protected ResponseData(Parcel in) {
            appointment = in.createTypedArrayList(Appointment.CREATOR);
            memberShip = in.createTypedArrayList(MemberShip.CREATOR);
        }


        @Override
        public int describeContents() {
            return 0;
        }

        @Override
        public void writeToParcel(Parcel dest, int i) {
            dest.writeTypedList(appointment);
            dest.writeTypedList(memberShip);
        }
    }

    public static class Appointment implements Parcelable {
        @SerializedName("InvoiceId")
        @Expose
        private String invoiceId;
        @SerializedName("InvoiceNumber")
        @Expose
        private String invoiceNumber;
        @SerializedName("Email")
        @Expose
        private String email;
        @SerializedName("Name")
        @Expose
        private String name;
        @SerializedName("NetAmount")
        @Expose
        private String netAmount;
        @SerializedName("Interval")
        @Expose
        private String interval;
        @SerializedName("Status")
        @Expose
        private String status;
        @SerializedName("InvoiceUrl")
        @Expose
        private String invoiceUrl;
        @SerializedName("Amount")
        @Expose
        private String amount;
        @SerializedName("Date")
        @Expose
        private String date;
        @SerializedName("InvoicePdf")
        @Expose
        private String invoicePdf;
        @SerializedName("InvoicePdfV")
        @Expose
        private String invoicePdfV;

        protected Appointment(Parcel in) {
            invoiceId = in.readString();
            invoiceNumber = in.readString();
            email = in.readString();
            name = in.readString();
            interval = in.readString();
            status = in.readString();
            invoiceUrl = in.readString();
            amount = in.readString();
            date = in.readString();
            invoicePdf = in.readString();
            invoicePdfV = in.readString();
            netAmount = in.readString();
        }

        public static final Creator<Appointment> CREATOR = new Creator<Appointment>() {
            @Override
            public Appointment createFromParcel(Parcel in) {
                return new Appointment(in);
            }

            @Override
            public Appointment[] newArray(int size) {
                return new Appointment[size];
            }
        };

        public String getInvoiceId() {
            return invoiceId;
        }

        public void setInvoiceId(String invoiceId) {
            this.invoiceId = invoiceId;
        }

        public String getInvoiceNumber() {
            return invoiceNumber;
        }

        public void setInvoiceNumber(String invoiceNumber) {
            this.invoiceNumber = invoiceNumber;
        }

        public String getInvoicePdf() {
            return invoicePdf;
        }

        public void setInvoicePdf(String invoicePdf) {
            this.invoicePdf = invoicePdf;
        }

        public String getInvoicePdfV() {
            return invoicePdfV;
        }

        public void setInvoicePdfV(String invoicePdfV) {
            this.invoicePdfV = invoicePdfV;
        }

        public static Creator<Appointment> getCREATOR() {
            return CREATOR;
        }

        public String getEmail() {
            return email;
        }

        public void setEmail(String email) {
            this.email = email;
        }

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }

        public String getInterval() {
            return interval;
        }

        public void setInterval(String interval) {
            this.interval = interval;
        }

        public String getStatus() {
            return status;
        }

        public void setStatus(String status) {
            this.status = status;
        }

        public String getInvoiceUrl() {
            return invoiceUrl;
        }

        public void setInvoiceUrl(String invoiceUrl) {
            this.invoiceUrl = invoiceUrl;
        }

        public String getAmount() {
            return amount;
        }

        public void setAmount(String amount) {
            this.amount = amount;
        }

        public String getDate() {
            return date;
        }

        public void setDate(String date) {
            this.date = date;
        }

        public String getNetAmount() {
            return netAmount;
        }

        public void setNetAmount(String netAmount) {
            this.netAmount = netAmount;
        }

        @Override
        public int describeContents() {
            return 0;
        }

        @Override
        public void writeToParcel(Parcel dest, int i) {
            dest.writeString(invoiceId);
            dest.writeString(invoiceNumber);
            dest.writeString(email);
            dest.writeString(name);
            dest.writeString(interval);
            dest.writeString(status);
            dest.writeString(invoiceUrl);
            dest.writeString(amount);
            dest.writeString(date);
            dest.writeString(invoicePdf);
            dest.writeString(invoicePdfV);
            dest.writeString(netAmount);
        }
    }

    public static class MemberShip implements Parcelable {
        @SerializedName("InvoiceId")
        @Expose
        private String invoiceId;
        @SerializedName("Email")
        @Expose
        private String email;
        @SerializedName("Name")
        @Expose
        private String name;
        @SerializedName("Interval")
        @Expose
        private String interval;
        @SerializedName("Status")
        @Expose
        private String status;
        @SerializedName("InvoiceUrl")
        @Expose
        private String invoiceUrl;
        @SerializedName("Amount")
        @Expose
        private String amount;
        @SerializedName("Date")
        @Expose
        private String date;
        @SerializedName("InvoicePdf")
        @Expose
        private String invoicePdf;
        @SerializedName("InvoicePdfV")
        @Expose
        private String invoicePdfV;

        protected MemberShip(Parcel in) {
            invoiceId = in.readString();
            email = in.readString();
            name = in.readString();
            interval = in.readString();
            status = in.readString();
            invoiceUrl = in.readString();
            amount = in.readString();
            date = in.readString();
            invoicePdf = in.readString();
            invoicePdfV = in.readString();
        }

        public static final Creator<MemberShip> CREATOR = new Creator<MemberShip>() {
            @Override
            public MemberShip createFromParcel(Parcel in) {
                return new MemberShip(in);
            }

            @Override
            public MemberShip[] newArray(int size) {
                return new MemberShip[size];
            }
        };

        public String getInvoiceId() {
            return invoiceId;
        }

        public void setInvoiceId(String invoiceId) {
            this.invoiceId = invoiceId;
        }

        public static Creator<MemberShip> getCREATOR() {
            return CREATOR;
        }

        public String getEmail() {
            return email;
        }

        public void setEmail(String email) {
            this.email = email;
        }

        public String getInvoicePdf() {
            return invoicePdf;
        }

        public void setInvoicePdf(String invoicePdf) {
            this.invoicePdf = invoicePdf;
        }

        public String getInvoicePdfV() {
            return invoicePdfV;
        }

        public void setInvoicePdfV(String invoicePdfV) {
            this.invoicePdfV = invoicePdfV;
        }

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }

        public String getInterval() {
            return interval;
        }

        public void setInterval(String interval) {
            this.interval = interval;
        }

        public String getStatus() {
            return status;
        }

        public void setStatus(String status) {
            this.status = status;
        }

        public String getInvoiceUrl() {
            return invoiceUrl;
        }

        public void setInvoiceUrl(String invoiceUrl) {
            this.invoiceUrl = invoiceUrl;
        }

        public String getAmount() {
            return amount;
        }

        public void setAmount(String amount) {
            this.amount = amount;
        }

        public String getDate() {
            return date;
        }

        public void setDate(String date) {
            this.date = date;
        }

        @Override
        public int describeContents() {
            return 0;
        }

        @Override
        public void writeToParcel(Parcel dest, int i) {
            dest.writeString(invoiceId);
            dest.writeString(email);
            dest.writeString(name);
            dest.writeString(interval);
            dest.writeString(status);
            dest.writeString(invoiceUrl);
            dest.writeString(amount);
            dest.writeString(date);
            dest.writeString(invoicePdf);
            dest.writeString(invoicePdfV);
        }
    }
}

File: app/src/main/java/com/brainwellnessspa/invoiceModule/models/InvoiceListModel.kt
package com.brainwellnessspa.invoiceModule.models

import android.os.Parcel
import android.os.Parcelable
import android.os.Parcelable.Creator
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.util.*

class InvoiceListModel {
    @SerializedName("ResponseData")
    @Expose
    var responseData: ResponseData? = null

    @SerializedName("ResponseCode")
    @Expose
    var responseCode: String? = null

    @SerializedName("ResponseMessage")
    @Expose
    var responseMessage: String? = null

    @SerializedName("ResponseStatus")
    @Expose
    var responseStatus: String? = null

    open class ResponseData protected constructor(`in`: Parcel) : Parcelable {
        @SerializedName("Appointment")
        @Expose
        var appointment: ArrayList<Appointment?>? = null

        @SerializedName("MemberShip")
        @Expose
        var memberShip: ArrayList<MemberShip?>? = null
        override fun describeContents(): Int {
            return 0
        }

        override fun writeToParcel(dest: Parcel, i: Int) {
            dest.writeTypedList(appointment)
            dest.writeTypedList(memberShip)
        }

        companion object {
            @JvmField val CREATOR: Creator<ResponseData?> = object : Creator<ResponseData?> {
                override fun createFromParcel(`in`: Parcel): ResponseData {
                    return ResponseData(`in`)
                }

                override fun newArray(size: Int): Array<ResponseData?> {
                    return arrayOfNulls(size)
                }
            }
        }

        init {
            appointment = `in`.createTypedArrayList(Appointment.CREATOR)
            memberShip = `in`.createTypedArrayList(MemberShip.CREATOR)
        }
    }

    open class Appointment protected constructor(`in`: Parcel) : Parcelable {
        @SerializedName("InvoiceId")
        @Expose
        var invoiceId: String? = `in`.readString()

        @SerializedName("InvoiceNumber")
        @Expose
        var invoiceNumber: String? = `in`.readString()

        @SerializedName("Email")
        @Expose
        var email: String? = `in`.readString()

        @SerializedName("Name")
        @Expose
        var name: String? = `in`.readString()

        @SerializedName("NetAmount")
        @Expose
        var netAmount: String? = `in`.readString()

        @SerializedName("Interval")
        @Expose
        var interval: String? = `in`.readString()

        @SerializedName("Status")
        @Expose
        var status: String? = `in`.readString()

        @SerializedName("InvoiceUrl")
        @Expose
        var invoiceUrl: String? = `in`.readString()

        @SerializedName("Amount")
        @Expose
        var amount: String? = `in`.readString()

        @SerializedName("Date")
        @Expose
        var date: String? = `in`.readString()

        @SerializedName("InvoicePdf")
        @Expose
        var invoicePdf: String? = `in`.readString()

        @SerializedName("InvoicePdfV")
        @Expose
        var invoicePdfV: String? = `in`.readString()

        override fun describeContents(): Int {
            return 0
        }

        override fun writeToParcel(dest: Parcel, i: Int) {
            dest.writeString(invoiceId)
            dest.writeString(invoiceNumber)
            dest.writeString(email)
            dest.writeString(name)
            dest.writeString(interval)
            dest.writeString(status)
            dest.writeString(invoiceUrl)
            dest.writeString(amount)
            dest.writeString(date)
            dest.writeString(invoicePdf)
            dest.writeString(invoicePdfV)
            dest.writeString(netAmount)
        }

        companion object {
            @JvmField val CREATOR: Creator<Appointment?> = object : Creator<Appointment?> {
                override fun createFromParcel(`in`: Parcel): Appointment {
                    return Appointment(`in`)
                }

                override fun newArray(size: Int): Array<Appointment?> {
                    return arrayOfNulls(size)
                }
            }
        }

    }

    open class MemberShip protected constructor(`in`: Parcel) : Parcelable {
        @SerializedName("InvoiceId")
        @Expose
        var invoiceId: String? = `in`.readString()

        @SerializedName("Email")
        @Expose
        var email: String? = `in`.readString()

        @SerializedName("Name")
        @Expose
        var name: String? = `in`.readString()

        @SerializedName("Interval")
        @Expose
        var interval: String? = `in`.readString()

        @SerializedName("Status")
        @Expose
        var status: String? = `in`.readString()

        @SerializedName("InvoiceUrl")
        @Expose
        var invoiceUrl: String? = `in`.readString()

        @SerializedName("Amount")
        @Expose
        var amount: String? = `in`.readString()

        @SerializedName("Date")
        @Expose
        var date: String? = `in`.readString()

        @SerializedName("InvoicePdf")
        @Expose
        var invoicePdf: String? = `in`.readString()

        @SerializedName("InvoicePdfV")
        @Expose
        var invoicePdfV: String? = `in`.readString()

        override fun describeContents(): Int {
            return 0
        }

        override fun writeToParcel(dest: Parcel, i: Int) {
            dest.writeString(invoiceId)
            dest.writeString(email)
            dest.writeString(name)
            dest.writeString(interval)
            dest.writeString(status)
            dest.writeString(invoiceUrl)
            dest.writeString(amount)
            dest.writeString(date)
            dest.writeString(invoicePdf)
            dest.writeString(invoicePdfV)
        }

        companion object {
            @JvmField val CREATOR: Creator<MemberShip?> = object : Creator<MemberShip?> {
                override fun createFromParcel(`in`: Parcel): MemberShip {
                    return MemberShip(`in`)
                }

                override fun newArray(size: Int): Array<MemberShip?> {
                    return arrayOfNulls(size)
                }
            }
        }

    }
}